pkg/resources/server_test.go

```diff
@@ -0,0 +1,327 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"github.com/Mellanox/k8s-rdma-shared-dev-plugin/pkg/types"
+	"github.com/Mellanox/k8s-rdma-shared-dev-plugin/pkg/types/mocks"
+	"github.com/Mellanox/k8s-rdma-shared-dev-plugin/pkg/utils"
+	"github.com/fsnotify/fsnotify"
+
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+	registerapi "k8s.io/kubernetes/pkg/kubelet/apis/pluginregistration/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/grpc"
+	"time"
+)
+
+type devPluginListAndWatchServerMock struct {
+	grpc.ServerStream
+}
+
+func (x *devPluginListAndWatchServerMock) Send(m *pluginapi.ListAndWatchResponse) error {
+	return nil
+}
+
+var _ = Describe("Server", func() {
+	Context("newResourcesServer", func() {
+		It("server with plugin watcher enabled", func() {
+			fs := utils.FakeFilesystem{
+				Dirs:     []string{"sys/class/net/ib0/"},
+				Symlinks: map[string]string{"sys/class/net/ib0/device": "../../../0000:02:00.0"},
```

Comment:
lets have these paths in a const at the top of the file

Reply:
add `fakeNetDevicePath` var